lib/report_generators/search_term_by_repo_and_searched_from.rb

```diff
@@ -247,4 +245,39 @@ def mean_ordinality_over_segments(clickthrough_segments)
 
       ordinality_fraction[:ordinality_total] / ordinality_fraction[:click_total]
   end
+
+private
+
+  def get_click_rows(analytics_client, start_index = 1, response_rows = [])
+    response = analytics_client.get_ga_data(@ga_profile_id, @start_date, @end_date, 'ga:totalEvents,ga:uniqueEvents,ga:avgEventValue', dimensions: 'ga:eventLabel,ga:eventAction,ga:dimension1,ga:dimension2', max_results: 10000, filters: "ga:eventCategory==Search;ga:eventAction==Clickthrough", sort: "ga:eventLabel", start_index: start_index)
+
+    @logger.info("Paginated find: #{response_rows.length + response.rows.length} of #{response.total_results} clicks")
+    # This is the last iteration, add these rows and return
+    if !response.next_link
+      response_rows.concat(response.rows)
+    else
+      next_index = CGI.parse(URI.parse(response.next_link).query)["start-index"].first
+      get_click_rows(analytics_client, next_index, response_rows.concat(response.rows))
```

Comment:
Nice! Like the iterative approach

Reply:
Oh yes - this kind of thing begs for...

# ReCuRsIoNNNNN!!!

![garyhandtw6it7lh2](https://user-images.githubusercontent.com/8731/33147479-e8996b3c-cf95-11e7-8035-242baa6f390a.gif)